### PR TITLE
add filtering for user input text

### PR DESCRIPTION
**Pull request by LucaVazz**
It may not be an issue in the community, but still "better safe than sorry".

This adds a (kind of aggressive but basic filtering) of all text which directly originates from user input and is used in SQL queries.

### CODE REVIEW

File: nyalib/small_helpers.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import re
+
+"""
+Intended for helper functions which are needed in multiple places but to small to warrant their own file.
+"""
+
+def filterTextForSqlInjection(text: str):
```
Comment (safaci2000) [inline nyalib/small_helpers.py:7]: This is not the right way of doing this.  We should use a prepared statement and avoid the need of escaping special characters. 

Unless the py driver doesn't support it this is an improvement, but we could do better.

**Comment by Yamitatsu** [comment]
some interesting insight here : https://stackoverflow.com/questions/1947750/does-python-support-mysql-prepared-statements

Seems like the way the query is sent is inherently safe.
Quite easily to find out though, let's just try to inject something ....

**Comment by LucaVazz** [comment]
As I understand the linked SO question and the source code of pymysqldb, they only encode characters into a common encoding not escape malicious characters like `;`.

So `!n.tag Gamer; DROP TABLE streams;` maybe causes problems as-is. But I'm not 100% sure, so someone should probably test it (in an isolated server).

**Comment by LucaVazz** [comment]
Turns out, I didn't look hard enough. pymtsql _does_ properly escape all malicious characters and wraps passed values into quotes, so all is already safe.

**Comment by Yamitatsu** [comment]
Oh, that's sweet !
Thank you very much @LucaVazz  for your input nonetheless, we learnt something very important thanks to you :)

**Comment by LucaVazz** [comment]
You're welcome :)

If you want to look into it more in-depth: cursor.execute calls functions which encode the passed parameters and pass them along to functions of connection. These are mainly wrappers /helpers around functions from the [converter-class](https://github.com/PyMySQL/PyMySQL/blob/master/pymysql/converters.py) in which the magic happens.

**Comment by Yamitatsu** [comment]
Yay! thank you :)

**Pull request closed without merging**